common/lib/xmodule/xmodule/js/src/capa/display.js

```diff
@@ -0,0 +1,1319 @@
+            this.reviewButton = this.$('.notification-btn.review-btn');
+            this.reviewButton.click(this.scroll_to_problem_meta);
+            this.submitButton = this.$('.action .submit');
+            this.submitButtonLabel = this.$('.action .submit .submit-label');
+            this.submitButtonSubmitText = this.submitButtonLabel.text();
+            this.submitButtonSubmittingText = this.submitButton.data('submitting');
+            this.submitButton.click(this.submit_fd);
+            this.hintButton = this.$('.action .hint-button');
+            this.hintButton.click(this.hint_button);
+            this.resetButton = this.$('.action .reset');
+            this.resetButton.click(this.reset);
+            this.showButton = this.$('.action .show');
+            this.showButton.click(this.show);
+            this.saveButton = this.$('.action .save');
+            this.saveNotification = this.$('.notification-save');
+            this.saveButton.click(this.save);
+            this.gentleAlertNotification = this.$('.notification-gentle-alert');
+            this.submitNotification = this.$('.notification-submit');
+
+            // Accessibility helper for sighted keyboard users to show <clarification> tooltips on focus:
+            this.$('.clarification').focus(function(ev) {
+                var icon;
+                icon = $(ev.target).children('i');
+                return window.globalTooltipManager.openTooltip(icon);
+            });
+            this.$('.clarification').blur(function() {
+                return window.globalTooltipManager.hide();
+            });
+            this.$('.review-btn').focus(function(ev) {
+                return $(ev.target).removeClass('sr');
+            });
+            this.$('.review-btn').blur(function(ev) {
+                return $(ev.target).addClass('sr');
+            });
+            this.bindResetCorrectness();
+            if (this.submitButton.length) {
+                this.submitAnswersAndSubmitButton(true);
+            }
+            Collapsible.setCollapsibles(this.el);
+            this.$('input.math').keyup(this.refreshMath);
+            if (typeof MathJax !== 'undefined' && MathJax !== null) {
+                this.$('input.math').each(function(index, element) {
+                    return MathJax.Hub.Queue([that.refreshMath, null, element]);
+                });
+            }
+        };
+
+        Problem.prototype.renderProgressState = function() {
+            var a, detail, earned, graded, possible, progress, progressTemplate, status;
+            detail = this.el.data('progress_detail');
+            status = this.el.data('progress_status');
+            graded = this.el.data('graded');
+
+            // Render 'x/y point(s)' if student has attempted question
+            if (status !== 'none' && (detail !== null && detail !== undefined) && (jQuery.type(detail) === 'string') &&
+                detail.indexOf('/') > 0) {
+                a = detail.split('/');
```

Comment:
After deeper look, there are many variable like `a` that are being used in this file, do you think we should change the names as part of this work ?